ConnectionFactory.php

```diff
@@ -45,8 +45,16 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        if (! isset($params['pdo']) && ! isset($params['charset'])) {
+        $connectionOverrideOptions = [];
+
+        if (isset($params['connection_override_options'])) {
+            $connectionOverrideOptions = $params['connection_override_options'];
+            unset($params['connection_override_options']);
+        }
+
+        if (! isset($params['pdo']) && (! isset($params['charset']) || (! empty($connectionOverrideOptions) && isset($params['url'])))) {
```

Comment:
Is that last `isset($params['url'])` needed? I mean, I get why you put it there... but I'm not sure it's strictly necessary.

Reply:
Added based on @stof feedback in a previous iteration https://github.com/doctrine/DoctrineBundle/pull/1290#discussion_r578595522